### PR TITLE
Replace struct update_fee_delta with lambda

### PR DESCRIPTION
**Pull request by maflcko**
The same was done for another struct in e177fcab3831b6d259da5164cabedcc9e78f6957.

Also, change type of feeDelta from int64_t to CAmount.

### CODE REVIEW

**Comment by hebasto** [comment]
Concept ACK.

File: src/txmempool.h [hunk 1]
```diff
@@ -131,7 +131,7 @@ class CTxMemPoolEntry
     std::chrono::seconds GetTime() const { return std::chrono::seconds{nTime}; }
     unsigned int GetHeight() const { return entryHeight; }
     int64_t GetSigOpCost() const { return sigOpCost; }
-    int64_t GetModifiedFee() const { return nFee + feeDelta; }
+    CAmount GetModifiedFee() const { return nFee + feeDelta; }
```
Comment (hebasto) [inline src/txmempool.h:134]: Also suggesting to add:
```diff
--- a/src/node/miner.h
+++ b/src/node/miner.h
@@ -45,7 +45,7 @@ struct CTxMemPoolModifiedEntry {
         nSigOpCostWithAncestors = entry->GetSigOpCostWithAncestors();
     }
 
-    int64_t GetModifiedFee() const { return iter->GetModifiedFee(); }
+    CAmount GetModifiedFee() const { return iter->GetModifiedFee(); }
     uint64_t GetSizeWithAncestors() const { return nSizeWithAncestors; }
     CAmount GetModFeesWithAncestors() const { return nModFeesWithAncestors; }
     size_t GetTxSize() const { return iter->GetTxSize(); }
```

**Review by hebasto — APPROVED** [review]
ACK fa131d3630bb8da6816e50a9a37c0ed4ad4ee33c, I have reviewed the code.

File: src/txmempool.h (hunk 1 above)
Comment (MarcoFalke) [reply src/txmempool.h:134]: Thx, done

**Review by hebasto — APPROVED** [review]
re-ACK fa84a49526fcf76e98b0b2f4d4b00b34a8dddf46

**Review by promag — COMMENTED** [review]
Code review ACK fa84a49526fcf76e98b0b2f4d4b00b34a8dddf46.

**Comment by promag** [comment]
Feel free to pull https://github.com/MarcoFalke/bitcoin-core/compare/2203-refactor-fee-delta-%F0%9F%94%B6...promag:2203-refactor-fee-delta-%F0%9F%94%B6

https://github.com/promag/bitcoin/commit/b440b03845410c2892f73497f774d29e897caed7
https://github.com/promag/bitcoin/commit/c5cdfe56de13fefb56f90d1af2f7912b0c19f1a6

**Comment by MarcoFalke** [comment]
I didn't touch the other structs, as they take multiple arguments. I think before touching them, we should ideally decide whether to use designated initializers or clang-tidy named arguments.

**Comment by fanquake** [comment]
@glozow want to review?

**Comment by glozow** [comment]
Concept ACK, haven't reviewed code

**Pull request merged**